core/src/main/java/org/apache/carbondata/core/carbon/metadata/schema/table/CarbonTable.java

```diff
@@ -504,4 +521,19 @@ public void setBlockSizeInMB(int blockSize) {
     this.blockSize = blockSize;
   }
 
+  /**
+   * to get the all dimension of a table
+   *
+   * @param tableName
+   * @return all dimension of a table
+   */
+  public CarbonDimension getAllDimensionByName(String tableName, String columnName) {
```

Comment:
This function is not getting all dimensions, it is used for finding the specified column, right? I think there is already a function called `getDimensionByName` to provide this functionality